utils/mof_checker/__init__.py

```diff
@@ -0,0 +1,123 @@
+import ipywidgets as ipw
+import functools as fct
+from traitlets import Instance, Int, List, observe
+from ase import Atoms, Atom
+from mofchecker import MOFChecker
+import tempfile
+from itertools import chain
+from IPython.display import clear_output
+
+
+ENABLED_CHECKS = [
+    "no_atomic_overlaps",
+    "no_undercoordinated_carbon",
+    "no_overcoordinated_carbon",
+    "no_overcoordinated_hydrogen",
+    "no_overcoordinated_nitrogen",
+    "no_undercoordinated_nitrogen",
+    "no_undercoordinated_rare_earth",
+    "no_floating_molecule",
+    "no_false_terminal_oxo",
+]
+
+class MofCheckerWidget(ipw.VBox):
+    """Widget that allows to check MOF structure correctness."""
+    structure = Instance(Atoms, allow_none=True)
+    selection = List(Int)
+
+    def __init__(self, title=''):
+        self.title = title
+        button_check = ipw.Button(description="Analyse structure")
+        button_check.on_click(self.check_structure)
+        self.checks = []
+        self._output = ipw.Output()
+        self.missing_h_button = ipw.Button(description="Add", layout={"width": "initial"})
+
+        super().__init__(children=[button_check,
+                                   self._output,
+                                  ])
+    
+    def check_structure(self, _=None):
+        
+        if self.structure is None:
+            return
+
+        self.mfchk = MOFChecker.from_ase(self.structure, primitive=False)
+        self.checks = []
+        if self.structure:
+            self.checks = [ check for (check_name, check) in self.mfchk.checks.items() if check_name in ENABLED_CHECKS]
+        
+        issue_found = False
+        
+        with self._output:
+            clear_output()
+            for check in self.checks:
+                if not check.is_ok:
+                    issue_found = True
+                    try:
+                        selection = list(chain.from_iterable(check.flagged_indices))
+                    except TypeError:
+                        selection = check.flagged_indices
+                    button = ipw.Button(description="Select", layout={"width": "initial"})
+                    button.on_click(fct.partial(self._select_atoms, selection=selection))
+                    #text = ipw.HTML("Found " + check.name.lower() + ": " + ', '.join(map(str, selection)))
+                    text = ipw.HTML("Found " + check.name.lower())
```

Comment:
if you have suggestions for better names, let me know I'm happy to change them!

Reply:
Not at the moment, but I know whom to contact in case I am not happy with a name :D